tensorflow/compiler/xla/service/cpu/ir_emitter.cc

```diff
@@ -1926,8 +1927,173 @@ Status IrEmitter::HandleSendDone(HloInstruction* send_done) {
   return Unimplemented("Send-done is not implemented on CPU.");
 }
 
-Status IrEmitter::HandleScatter(HloInstruction*) {
+Status IrEmitter::HandleScatter(HloInstruction* scatter) {
+  // (TODO) this is to prevent the crash untill the porting is complete
   return Unimplemented("Scatter is not implemented on CPUs.");
+  // ------------
+  const HloInstruction* operand = scatter->operand(0);
+  const HloInstruction* scatter_indices = scatter->operand(1);
+  const HloInstruction* updates = scatter->operand(2);
+
```

Comment:
This was a pre-mlir GPU emitter porting from (git tag 2.3.4).
It seem that the HLO signature change over the time and so the mlir version of the algo is changed on master and it has more parameters:

```c++
      ScatterDescriptor desc;
      desc.name = IrName(root);
      desc.operand_shape = root->operand(0)->shape();
      desc.scatter_indices_shape = root->operand(1)->shape();
      desc.updates_shape = updates_shape;
      desc.dim_numbers = dim_numbers;
      desc.unique_indices = root->unique_indices();
      desc.update_computation = root->called_computations()[0];
      desc.output = ir_arrays.back();
```